### PR TITLE
Skeleton for Environment

### PR DESCRIPTION
**Pull request by zpatrick**
This includes a bunch of misc changes for how dealing with environment entities will work. 

I've cleared out a bunch of scheduler code so we can re-visit that once the other entities are done.

### CODE REVIEW

File: api/controllers/controller.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package controllers
```
Comment (diemonster) [inline api/controllers/controller.go]: just noting that the relevant change here is renaming the previous `handlers` package to `controllers`

File: api/controllers/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package controllers
```
Comment (diemonster) [inline api/controllers/controller_test.go]: Far more simple than the previous handler tests, I definitely prefer this.

File: api/controllers/environment.go [hunk 3]
```diff
@@ -28,7 +32,7 @@ func (e *EnvironmentController) CreateEnvironment(request *restful.Request, resp
 		// todo: handle error
 	}
 
-	environmentModel, err := environment.Read()
+	environmentModel, err := environment.Model()
```
Comment (diemonster) [inline api/controllers/environment.go]: [nit] maybe `GetModel` versus `Model`?

Comment (diemonster) [inline api/controllers/environment.go]: ahhh I see why you did `Model` when put in context with the other `Get` methods....

File: api/controllers/environment.go [hunk 4]
```diff
@@ -37,10 +41,55 @@ func (e *EnvironmentController) CreateEnvironment(request *restful.Request, resp
...
+	WriteJobResponse(response, jobID)
 }
 
 func (e *EnvironmentController) GetEnvironment(request *restful.Request, response *restful.Response) {
+	id := request.PathParameter("id")
+	if id == "" {
+		// todo: handle error
+	}
+
+	environment := e.Provider.GetEnvironment(id)
```
Comment (diemonster) [inline api/controllers/environment.go]: Discussions about the ambiguity of `Provider` aside, this is pretty clean from the calling code's point of view. slightly confusing to me to see an abstract object returned from `GetEnvironment` and then a model returned from just `Model`...

File: common/models/job.go [hunk 5]
```diff
@@ -2,13 +2,15 @@ package models
...
+
+	"github.com/quintilesims/layer0/common/job"
 )
 
 type Job struct {
 	JobID       string            `json:"job_id"`
 	TaskID      string            `json:"task_id"`
-	JobStatus   int64             `json:"job_status"`
-	JobType     int64             `json:"job_type"`
+	JobStatus   job.JobStatus     `json:"job_status"`
```
Comment (diemonster) [inline common/models/job.go]: would this be a breaking change given that this is now a string?

File: api/controllers/environment.go (hunk 4 above)
Comment (zpatrick) [inline api/controllers/environment.go]: Open to using different verb if we can find one. Don't think this should be a blocker for this pr though

File: common/models/job.go (hunk 5 above)
Comment (zpatrick) [inline common/models/job.go]: It's still an int64

Comment (diemonster) [inline common/models/job.go]: opps, disregard!

File: api/controllers/environment.go (hunk 4 above)
Comment (diemonster) [inline api/controllers/environment.go]: Best I can think of for now is `InitModel` or `NewModel`, I agree it's not a blocker for the PR

**Pull request merged**